.github/workflows/composite/bazel-gh-cache/action.yml

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Update Bazel Caches on GitHub
+description: Pulls caches for .bazel-cache and .bazel-cache-repo for a given key prefix and purges them based on a threshold
+
+inputs:
+  cache-key-prefix:
+    description: A prefix used for cache keys
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Set variables
+      shell: bash
+      run: |
+        echo "BAZEL_CACHE_NAME=${{ env.BAZEL_CACHE_NAME }}" >> $GITHUB_ENV
+        echo "BAZEL_CACHE_REPO_NAME=${{ env.BAZEL_CACHE_REPO_NAME }}" >> $GITHUB_ENV
+        echo "BAZEL_CACHE_CUTOFF_MB=${{ env.BAZEL_CACHE_CUTOFF_MB }}" >> $GITHUB_ENV
+        echo "BAZEL_CACHE_REPO_CUTOFF_MB=${{ env.BAZEL_CACHE_REPO_CUTOFF_MB }}" >> $GITHUB_ENV
+        echo "BAZEL_CACHE_DIR=.${{ env.BAZEL_CACHE_NAME }}" >> $GITHUB_ENV
+        echo "BAZEL_CACHE_REPO_DIR=.${{ env.BAZEL_CACHE_REPO_NAME }}" >> $GITHUB_ENV
+      env:
+        BAZEL_CACHE_NAME: bazel-cache
+        BAZEL_CACHE_REPO_NAME: bazel-cache-repo
+        BAZEL_CACHE_CUTOFF_MB: 500
```

Comment:
Note: hard coded for now. If this action ever gets used for other use cases then the API can be extended to allow configurable thresholds.

Reply:
If I read the logging correctly the repo cache is ~200 MB and the bazel cache is ~30 MB - if this is the case then we could probably limit the bazel cache to 100 or 200 MB.

Reply:
I would not optimize that much - imho 500+500mb=1gb is reasonable for our current usage of the GitHub caches.